crates/astria-composer/src/types/searcher.rs

```diff
@@ -0,0 +1,39 @@
+//! TODO: explain searcher engine & types
+use std::pin::Pin;
+
+use async_trait::async_trait;
+use color_eyre::eyre;
+use tokio_stream::Stream;
+
+/// A stream of events emitted by a [Collector](Collector).
+// TODO
+pub type CollectorStream<'a, E> = Pin<Box<dyn Stream<Item = E> + Send + 'a>>;
+
+/// Collector trait, which defines a source of events.
+#[async_trait]
+pub trait Collector<E>: Send + Sync {
```

Comment:
I know it's very enticing to start out with high level abstractions, but I would recommend starting concrete `collector`, `strategy`, and `executor` types (just one of each), and then going from there.

Chances are, you will never need the abstractions. And also you will have a better understanding of what it is you are trying to abstract away, reducing the chances of your abstractions being leaky: https://en.wikipedia.org/wiki/Leaky_abstraction

Reply:
agreed, this was copied over from [atremis](https://github.com/paradigmxyz/artemis/blob/main/crates/artemis-core/src/types.rs) when i started the repo but don't think i need to use it so removed.